# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/dns/nsHostResolver.cpp
# Commit: 4018c85570aa
# Full Hash: 4018c85570aa474bb68199e5a068c48382254fb7
# Author: Valentin Gosu <valentin.gosu@gmail.com>
# Date: 2021-03-25 03:42:26
# Description:
#   Bug 1700091 - Always hold mutex when accessing TypeHostRecord::mResults r=necko-reviewers,kershaw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D109343
# ==============================================================================

diff -r 228d6b490775 -r 4018c85570aa netwerk/dns/nsHostResolver.cpp
--- a/netwerk/dns/nsHostResolver.cpp	Wed Mar 24 14:11:49 2021 +0000
+++ b/netwerk/dns/nsHostResolver.cpp	Wed Mar 24 14:17:48 2021 +0000
@@ -629,6 +629,7 @@
 NS_IMETHODIMP
 TypeHostRecord::GetHasIPAddresses(bool* aResult) {
   NS_ENSURE_ARG(aResult);
+  MutexAutoLock lock(mResultsLock);
 
   if (!mResults.is<TypeRecordHTTPSSVC>()) {
     return NS_ERROR_NOT_AVAILABLE;
@@ -642,6 +643,7 @@
 NS_IMETHODIMP
 TypeHostRecord::GetAllRecordsExcluded(bool* aResult) {
   NS_ENSURE_ARG(aResult);
+  MutexAutoLock lock(mResultsLock);
 
   if (!mResults.is<TypeRecordHTTPSSVC>()) {
     return NS_ERROR_NOT_AVAILABLE;
